source/Target/SwiftLanguageRuntime.cpp

```diff
@@ -2479,59 +2418,89 @@ bool SwiftLanguageRuntime::GetDynamicTypeAndAddress(
     TypeAndOrName &class_type_or_name, Address &address,
     Value::ValueType &value_type) {
   class_type_or_name.Clear();
+  if (use_dynamic == lldb::eNoDynamicValues || !CouldHaveDynamicValue(in_value))
+    return false;
 
-  if (SwiftASTContext *swift_ast = llvm::dyn_cast_or_null<SwiftASTContext>(
-          in_value.GetCompilerType().GetTypeSystem())) {
-    if (swift_ast->HasFatalErrors() || !swift_ast->GetClangImporter()) {
-      return false;
+  // Dynamic type resolution in RemoteAST might pull in other Swift modules, so
+  // use the scratch context where such operations are legal and safe.
+  assert(IsScratchContextLocked(in_value.GetTargetSP()) &&
+         "Swift scratch context not locked ahead of dynamic type resolution");
+  auto scratch_ctx = in_value.GetScratchSwiftASTContext();
+  if (!scratch_ctx)
```

Comment:
Can we actually get here? We just asserted that we locked a scratchASTContext so this check seems redundant?

Reply:
We can get there if target==nullptr (not sure if that is possible, but perhaps when evaluating expressions without creating a target) or if the valueobject doesn't have a valid module.

Either of those cases seem really unlikely, but it I can't prove they are impossible.

Reply:
Fair, better safe than sorry (here I think it's better being defensive).